src/client/db.rs

```diff
@@ -0,0 +1,109 @@
+use bson;
+use bson::Bson;
+use client::MongoClient;
+use client::coll::Collection;
+use client::common::{ReadPreference, WriteConcern};
+
+/// Interfaces with a MongoDB database.
+pub struct Database<'a> {
+    pub name: String,
+    pub client: &'a MongoClient,
+    read_preference: Option<ReadPreference>,
```

Comment:
Why are `read_preference` and `write_concern` members attached to the `Database`?  A write concern is enforced when doing writes (or when executing commands that could do writes such as write commands and findAndModify -- for 3.2 -- which I incidentally added to the server :) ). Think about what having a `Database` have a read preference and a write concern means. As an exercise, take a look at how other drivers use these fields and remove these fields for now. But I want you to start thinking about read preference and write concern.